waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/PrefixMapping.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2016-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PrefixMapping extends MetaStoreMappingDecorator {
```

Comment:
It seems to me that this is the actual decorator and that `MetastoreMappingDecorator` is just delegating the calls but not decorating anything.

Reply:
yeah I think that's how you write it though you have an abstract "decorator" class and then concrete subclasses. But my head first java design patterns has been a while :)

Reply:
Yes, your implementation is spot on and just how it's recommended in the GoF book. Having the abstract class is useful if you've got many decorators. If you don't, you can just simplify it and have a single implementation that decorates.

Reply:
we have 2 concrete subclasses

Reply:
I missed the other one then. Thanks.